### PR TITLE
fix(docs-infra): fix sidenav elements being focusable off screen

### PR DESCRIPTION
**Pull request by dario-piotrowicz**
some elements in the aio sidenav (especially anchors) are still
included in the tab order even though they are not visible, fix
such behavior by applying `display: none` to the closed sidenav
so that its elements become non-interactable

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [x] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

In the aio sidevan elements off screen are still present in the tab order which is not correct

For example:
sidenav-before
(notice the urls at the bottom of the page appearing as I am pressing `tab`)

Issue Number: N/A

## What is the new behavior?

I've added a display none to the sidenav so that its elements cannot get focus anymore:
sidenav-after

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/src/styles/1-layouts/sidenav/_sidenav.scss [hunk 1]
```diff
@@ -33,6 +33,10 @@ mat-sidenav-container.sidenav-container {
     left: 0;
     min-width: 260px;
 
+    &:not(.mat-drawer-opened) {
```
Comment (dario-piotrowicz) [inline aio/src/styles/1-layouts/sidenav/_sidenav.scss:36]: @gkalpak before you say anything I just wanted to say that I think `:not` is now ok to use :smile: ([caniuse link](https://caniuse.com/?search=%3Anot()))

File: aio/src/app/app.component.ts [hunk 2]
```diff
@@ -54,7 +54,7 @@ export class AppComponent implements OnInit {
    */
   pageId: string;
   /**
-   * An HTML friendly identifer for the "folder" of the currently displayed page.
+   * An HTML friendly identifier for the "folder" of the currently displayed page.
```
Comment (dario-piotrowicz) [inline aio/src/app/app.component.ts:57]: I'm piggybacking an unrelated typo fix here, I hope that it's fine :sweat_smile: 
(it does have a separate commit :slightly_smiling_face: )

File: aio/src/styles/1-layouts/sidenav/_sidenav.scss [hunk 3]
```diff
@@ -33,6 +33,10 @@ mat-sidenav-container.sidenav-container {
...
+      display: none;
```
Comment (crisbeto) [inline aio/src/styles/1-layouts/sidenav/_sidenav.scss:37]: This shouldn't be necessary with the Material sidenav since we set `visibility: hidden` when it's off-screen which also prevents focus. Maybe one of the child elements has `visibility: visible` which overrides the one coming from the parent?

Comment (dario-piotrowicz) [reply aio/src/styles/1-layouts/sidenav/_sidenav.scss:37]: ah... yeah good point, I'll check it out thanks! :slightly_smiling_face::+1:

Comment (dario-piotrowicz) [reply aio/src/styles/1-layouts/sidenav/_sidenav.scss]: @crisbeto I've fixed the visibility, thank you so very much you comment was totally on point! :heart_eyes: 

(PS: sorry about my previous solution :bow: :cry:)

File: aio/src/styles/1-layouts/sidenav/_sidenav.scss [hunk 4]
```diff
@@ -101,7 +101,6 @@ aio-nav-menu {
 
     .heading-children {
       &.expanded {
-        visibility: visible;
```
Comment (dario-piotrowicz) [inline aio/src/styles/1-layouts/sidenav/_sidenav.scss:104]: I've removed this as it doesn't seem necessary and makes the elements interactable by overriding the mat-sidenav `visibility: hidden` (as suggested by @crisbeto [here](https://github.com/angular/angular/pull/44990#discussion_r800421767))

this was most likely just added to counter the `visibility: hidden` of the `.collapsed` version (as you can see below, on line 114/113)

when I've removed it I checked the navbar and all seems to be still working as expected, so it should be all good :smile:

File: aio/src/styles/1-layouts/sidenav/_sidenav.scss (hunk 3 above)
Comment (crisbeto) [reply aio/src/styles/1-layouts/sidenav/_sidenav.scss]: Here's an upstream fix: https://github.com/angular/components/pull/24376

**Comment by dario-piotrowicz** [comment]
Closing as no longer necessary as the bug will be fixed by [@crisbeto's PR](https://github.com/angular/components/pull/24376)

**Comment by crisbeto** [comment]
@dario-piotrowicz I mentioned my PR mostly for reference. I can't guarantee when it'll actually be merged since it may cause issues internally. It'll also take some time to make it into a release that can be used by AIO. It may be worth merging this PR in and then reverting it once the fix is available.

**Comment by dario-piotrowicz** [comment]
> @dario-piotrowicz I mentioned my PR mostly for reference. I can't guarantee when it'll actually be merged since it may cause issues internally. It'll also take some time to make it into a release that can be used by AIO. It may be worth merging this PR in and then reverting it once the fix is available.

Ah ok, sorry I jumped the gun there! :sweat_smile: 

PR reopened, sorry! :sweat_smile: :bow:

**Review by gkalpak — APPROVED** [review]
LGTM :+1:
Thx!

**Comment by dario-piotrowicz** [comment]
Closing this PR as the issue is no longer present in aio (thanks to @crisbeto's PR :slightly_smiling_face:)
